doc/source/conf.py

```diff
@@ -74,6 +74,7 @@
               'sphinx.ext.ifconfig',
               'sphinx.ext.linkcode',
               'nbsphinx',
+              'flake8_rst.sphinxext.custom_roles',
```

Comment:
What does this do? I know we haven't done that before, but would be good to have a comment

Reply:
flake8-rst allows for customized merging code-blocks. Like
```
.. code-block:: python
   :flake8-group: GroupForExample#1

   code...
```
However sphinx throws an exception if a directive is decorated by an unknown role.
So all it does is telling the IPython and code-block directive that certain roles can be added.

Reply:
Would it be a problem for you postponing this to a new PR? I guess that's the best solution, but would like to give it a second thought, as I'm considering to build the docs without sphinx (#23763), and everything that requires a sphinx plugin would be extra work if we do that.

If it's not giving you more work or headaches, I'd prefer to fix everything else for now (may be we can ignore `F821` which I think it's the main error avoided by this plugin), and then be back to this. Would that be ok?